client/castApp/src/castinit.c

```diff
@@ -65,13 +67,115 @@ static void casthook(initHookState state)
     epicsAtExit(&castexit, NULL);
 }
 
+/*
+    Example call: addReccasterEnvVars("SECTOR") or addReccasterEnvVars("SECTOR", "BUILDING")
+    Appends the given env variables to the extra_envs list to be sent in addition to the default_envs array
+*/
+static void addReccasterEnvVars(caster_t* self, int argc, char **argv)
+{
+    int i, j;
+    int ret = 0;
+    if(argc < 2) {
+        errlogSevPrintf(errlogMinor, "At least one argument expected for %s\n", __func__);
+        return;
+    }
+    /* sanitize input - check for dups and empty args. First arg in argv is function name so skip that */
+    int argCount = argc - 1;
+    epicsMutexMustLock(self->lock);
+    for(i = 1; i < argc; i++) {
+        if(argv[i] == NULL) {
+            argCount--;
+            errlogSevPrintf(errlogMinor, "Arg is NULL for %s\n", __func__);
+            continue;
+        }
+        else if(argv[i][0] == '\0') {
+            argCount--;
+            errlogSevPrintf(errlogMinor, "Arg is empty for %s\n", __func__);
+            argv[i] = NULL;
+            continue;
+        }
+        /* check if dup in self->extra_envs. doesn't check if arg is in default_envs right now */
+        for(j = 0; j < self->num_extra_envs; j++) {
+            if(strcmp(argv[i], self->extra_envs[j]) == 0) {
+              argCount--;
+              errlogSevPrintf(errlogMinor, "Env var %s is already in extra_envs list\n", argv[i]);
+              argv[i] = NULL;
+              break;
+            }
+        }
+    }
+    epicsMutexUnlock(self->lock);
+
+    char ** tmp_new_envs;
+    tmp_new_envs = calloc(argCount, sizeof(*tmp_new_envs));
+    if(tmp_new_envs == NULL) {
+        errlogSevPrintf(errlogMajor, "Error in memory allocation of tmp_new_envs from %s\n", __func__);
+        return;
+    }
+    for(i = 0, j = 0; i < argc - 1; i++) {
+        /* check for bad args which are set to NULL above */
+        if(!argv[i+1])
+            continue;
+        if((tmp_new_envs[j] = strdup(argv[i+1])) == NULL) {
+            ret = 1;
+        }
+        j++;
+    }
```

Comment:
Very minor nitpick, but this for loop runs from 0 to `argc - 1` (with index `i+1`) while the one above runs from 1 to `argc` (with index `i`). It might be nicer to harmonise this...

Reply:
agreed. I made a note of this and add it in with the remaining cleanup most likely that is needed

Reply:
updated these loops to match